core/shared/src/main/scala/zio/ZIO.scala

```diff
@@ -400,6 +400,9 @@ sealed trait ZIO[-R, +E, +A] extends Serializable { self =>
   final def forkOn(ec: ExecutionContext): ZIO[R, E, Fiber[E, A]] =
     self.on(ec).fork
 
+  final def fuseError[E1, E2 <: E1](default: E2)(implicit ev: E <:< Option[E1]): ZIO[R, E1, A] =
```

Comment:
How about we call this `flattenErrorOption`?

Reply:
Definitely more descriptive.